### PR TITLE
binfmt: Move [elf|nxflat]_[un]initialize to private header file

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

like what builtin binary format do

## Impact

code refactor only

## Testing

ci

### CODE REVIEW

File: binfmt/binfmt.h [hunk 1]
```diff
@@ -203,4 +203,71 @@ void builtin_uninitialize(void);
...
+ *
+ * Description:
+ *   Unregister the ELF binary loader
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ELF
```
Comment (pkarashchenko) [inline binfmt/binfmt.h:235]:
```suggestion

/****************************************************************************
 * Name: elf_uninitialize
 *
 * Description:
 *   Unregister the ELF binary loader
 *
 * Returned Value:
 *   None
 *
 ****************************************************************************/

```

File: binfmt/binfmt.h [hunk 2]
```diff
@@ -203,4 +203,71 @@ void builtin_uninitialize(void);
...
+ *
+ * Description:
+ *   Unregister the NXFLAT binary loader
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_NXFLAT
```
Comment (pkarashchenko) [inline binfmt/binfmt.h:269]:
```suggestion

/****************************************************************************
 * Name: nxflat_uninitialize
 *
 * Description:
 *   Unregister the NXFLAT binary loader
 *
 * Returned Value:
 *   None
 *
 ****************************************************************************/

```

**Review by pkarashchenko — APPROVED** [review]

File: binfmt/binfmt.h (hunk 2 above)
Comment (xiaoxiang781216) [reply binfmt/binfmt.h]: Done.

File: binfmt/binfmt.h (hunk 1 above)
Comment (xiaoxiang781216) [reply binfmt/binfmt.h]: Done.

File: libs/libc/modlib/modlib_symbols.c [hunk 3]
```diff
@@ -527,11 +528,11 @@ int modlib_insertsymtab(FAR struct module_s *modp,
...
  ****************************************************************************/
 
-static int findEP(const void *c1, const void *c2)
+static int findep(const void *c1, const void *c2)
 {
-  const struct epTable_s *m1 = (struct epTable_s *) c1;
-  const struct epTable_s *m2 = (struct epTable_s *) c2;
-  return strcmp((FAR const char *)m1->epName, (FAR const char *)m2->epName);
+  const struct eptable_s *m1 = (struct eptable_s *)c1;
+  const struct eptable_s *m2 = (struct eptable_s *)c2;
```
Comment (pkarashchenko) [inline libs/libc/modlib/modlib_symbols.c:534]:
```suggestion
  FAR const struct eptable_s *m1 = (FAR struct eptable_s *)c1;
  FAR const struct eptable_s *m2 = (FAR struct eptable_s *)c2;
```

File: libs/libc/modlib/modlib_symbols.c [hunk 4]
```diff
@@ -527,11 +528,11 @@ int modlib_insertsymtab(FAR struct module_s *modp,
  *
  ****************************************************************************/
 
-static int findEP(const void *c1, const void *c2)
+static int findep(const void *c1, const void *c2)
```
Comment (pkarashchenko) [inline libs/libc/modlib/modlib_symbols.c:531]:
```suggestion
static int findep(FAR const void *c1, FAR const void *c2)
```

File: libs/libc/modlib/modlib_symbols.c [hunk 5]
```diff
@@ -549,23 +550,23 @@ void *modlib_findglobal(FAR struct module_s *modp,
                         struct mod_loadinfo_s *loadinfo,
```
Comment (pkarashchenko) [inline libs/libc/modlib/modlib_symbols.c:550]:
```suggestion
                        FAR struct mod_loadinfo_s *loadinfo,
```

File: libs/libc/modlib/modlib_symbols.c [hunk 6]
```diff
@@ -549,23 +550,23 @@ void *modlib_findglobal(FAR struct module_s *modp,
...
 {
-  FAR Elf_Shdr *strTab = &loadinfo->shdr[shdr->sh_link];
+  FAR Elf_Shdr *strtab = &loadinfo->shdr[shdr->sh_link];
   int ret;
-  struct epTable_s key;
-  struct epTable_s *res;
-  extern struct epTable_s globalTable[];
-  extern int nGlobals;
+  struct eptable_s key;
+  struct eptable_s *res;
```
Comment (pkarashchenko) [inline libs/libc/modlib/modlib_symbols.c:556]:
```suggestion
  FAR struct eptable_s *res;
```

File: libs/libc/modlib/modlib_symbols.c [hunk 7]
```diff
@@ -549,23 +550,23 @@ void *modlib_findglobal(FAR struct module_s *modp,
...
+  extern struct eptable_s global_table[];
+  extern int nglobals;
```
Comment (pkarashchenko) [inline libs/libc/modlib/modlib_symbols.c:558]: Can we move externs to the beginning of the file?

File: libs/libc/modlib/modlib_symbols.c [hunk 8]
```diff
@@ -549,23 +550,23 @@ void *modlib_findglobal(FAR struct module_s *modp,
...
 
-  ret = modlib_symname(loadinfo, sym, strTab->sh_offset);
+  ret = modlib_symname(loadinfo, sym, strtab->sh_offset);
   if (ret < 0)
       return NULL;
```
Comment (pkarashchenko) [inline libs/libc/modlib/modlib_symbols.c:562]:
```suggestion
    {
      return NULL;
    }
```

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**